### DISCUSSION TITLE
Do we need a timed leash release?

### DISCUSSION THREAD

**Discussion by SilkieSabra** [OpenCollarTeam/OpenCollar#985] [Polls]
In OpenCollar for a long time, the leash stays active if the leasher leaves the area, whether that means logging off or moving out of range.  If the leasher has owner access to the collar, this means that anyone with a lower level of access can't grab the leash unless the wearer safewords.  Should this be corrected?  Relates to Issue #982 .

**Comment by catachrest** [comment]
If it's not insane to code, my preference would be having the timer with a relatively short time (eg 5-10 min) should be the default but with the option for the leasher to set a longer timer or to turn the timer off to have the leash remain in force until they get back. It might also be useful to have different settings for when you're leashed to the person who then leaves vs  being anchored to a post or other object.

**Reply by Surssin** [reply]
Many systems do just this. If you're posted, the leash never drops, because, well, the post is always there to hold you. Otherwise, if the person holding your leash vanishes, after a set time, it drops.

Though they don't have the customization options mentioned. Potentially doable, but tricky.

**Comment by Medea-Destiny** [comment]
I think the better solution to this is to allow for lower auth levels to use the leash while a higher auth level who has leashed the collar wearer is not present. This could be achieved by maintaining two keys, one for the current leash holder, and one for the highest auth leash holder who has not yet released the collar.

**Reply by Surssin** [reply]
This can be dicey though, especially when you factor in posting. It may be REALLY hard to code in a separate set of checks for when a higher auth tied the leash to a post. You wouldn't want a lower auth to be able to untie the sub from a post.

Or perhaps not. I can't say. That's a level of coding beyond me. It's an interesting idea.

**Reply by Medea-Destiny** [reply]
Easy enough, we just check whether the avatar is present if leashed to an avatar, or go purely on auth if leashed to an object

**Reply by Thanars** [reply]
Being leashed to a person who then leaves or being leashed to a post are very different things. One is either an accident/oversight or very temporary, the other is intentional.
For the former I agree with Medea, just let others take the leash.
When leashing to a post or something, an automatic, configurable timer would be great. When the timer is up give the option to unleash or restart the timer.

Edit: Didn't realize how old this topic is..
